docs/api/vulnerability-info.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Vulnerability Info, NuGet API
+description: The data that allows clients to check packages for known vulnerabilities.
+author: zivkan
+ms.author: zivkan
+ms.date: 06/16/2023
+ms.topic: reference
+---
+
+# Vulnerability information
+
+NuGet client, starting from version 6.7, can download known package vulnerability information to use in scenarios such as checking packages during restore operations.
+
+The API consists of at least two files, the [vulnerability index](#vulnerability-index) and one or more [vulnerability data files](#vulnerability-data).
+Known vulnerability data can be partitioned into multiple files, and the vulnerability index provides clients with information needed to cache files, and update the cache, efficiently.
+
+The resource used for building this URL is the `VulnerabilityInfo` resource found in the [service index](service-index.md).
+
+## Versioning
+
+The following `@type` values are used:
+
+@type value                       | Notes
+--------------------------------- | -----
+VulnerabilityInfo/6.7.0 | The initial release
+
+## Suggested partitioning strategy
+
+The files listed in the vulnerability index should ideally be optimized to maximize caching, and therefore minimize updates to large files.
+This will allow clients to minimize the frequency it needs to download updates.
+
+A suggested strategy for vulnerability data partitioning is to have two files, `base.json` and `updates.json`.
```

Comment:
Should we mention about whether the files should be compressed or not? What kind of compressions does NuGet clients support?

Reply:
As long as the server respects the the `Accept-Encoding` header on requests, then client will "Just Work". This also allows servers to adapt as new versions of .NET's runtime support newer compression formats

Note, I wasn't around when the registration resource added support for gzip, so I don't know why the gzipped versions are in a different URL to the non-gzipped versions. The URL scheme is the same (client doesn't append `.gz` to the request), so as long as the server handles requests without the `Accept-Encoding` header to request the uncompressed file.

So, I see this as a generic HTTP compliance issue, not something specific to the VulnerabilityInfo resource.

Reply:
In the package metadata endpoint, we have some resources returning gzip and others not. Notably, `Accept-Encoding` is NOT guaranteed to be respected by the server.

On the other hand, the package content resource does not talk about gzip at all.

I think this is more about client flexibility than server respecting request headers. The server can return gzipped JSON or not and official client will handle it because it is configured to do so.

We can choose to remain vague here, knowing that server will never do anything that breaks client (hopefully, it's our job after all) and that 3rd party client implementations can figure this out pretty easily. Or we can be explicit and go into what is happening on NuGet.org in practice. I don't feel strongly personally.

Reply:
> In the package metadata endpoint, we have some resources returning gzip and others not. Notably, Accept-Encoding is NOT guaranteed to be respected by the server.

👀 😱 

This is completely unintuitive to me, but I wasn't on the NuGet team at the time this was implemented, so I don't know why this decision was made.

Reply:
It was done this way because Azure Blob Storage doesn't respect the Accept-Encoding request header. It just serves whatever `Content-Encoding` is baked into the blob.